SushiBeltTrackerScrollContext.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  SushiBeltTrackerScrollContext.swift
+//  SushiBelt
+//
+//  Created by Jaxtyn on 2022/08/16.
+//
+
+import Foundation
+
+#if !os(macOS)
+
+import UIKit
+
+typealias PlatformScrollView = UIScrollView
+
+#else
+
+import AppKit
+
+typealias PlatformScrollView = NSScrollView
+
+#endif
+
+public protocol SushiBeltTrackerScrollContext: AnyObject {
+  func scrollDrection() -> SushiBeltTrackerScrollDirection?
+}
+
+extension PlatformScrollView: SushiBeltTrackerScrollContext {
+  public func scrollDrection() -> SushiBeltTrackerScrollDirection? {
```

Comment:
I think, `internal` access control is enough!

Reply:
oh. I just see that SushiBeltTrackerScrollContext is public protocol.
well... How about support a default UIScrollView base Context replace from default extension?
for example. 
```swift
public final class SushiBeltTrackerUIScrollContext: SushiBeltTrackerScrollContext {
    weak var scrollView: UIScrollView?
    public init(scrollView: UIScrollView) { ... }
    public func scrollDrection() -> SushiBeltTrackerScrollDirection? { ... }
}
```

Reply:
good opinion applied it
thx :)

https://github.com/daangn/SushiBelt/pull/9/commits/0510e034e7b772dc979225bc7a7b886bdf0bb3e6